ethers-solc/src/artifacts/mod.rs

```diff
@@ -494,6 +495,35 @@ impl Default for Settings {
     }
 }
 
+impl Hash for Settings {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.optimizer.enabled.hash(state);
+        self.optimizer.runs.hash(state);
+        if let Some(SettingsMetadata { bytecode_hash, cbor_metadata, .. }) = self.metadata {
+            (bytecode_hash.unwrap_or_default() as u8).hash(state);
+            cbor_metadata.hash(state);
+        }
+        self.output_selection.0.keys().cloned().collect::<String>().hash(state);
+        (self.evm_version.unwrap_or_default() as u8).hash(state);
+        self.via_ir.hash(state);
+        if let Some(DebuggingSettings { revert_strings, debug_info, .. }) = &self.debug {
+            (revert_strings.unwrap_or_default() as u8).hash(state);
+            debug_info.hash(state);
+        }
+    }
+}
+
+impl PartialEq for Settings {
```

Comment:
what's the reason for the manual impls here?

manual hash impl for settings and including the metadatahash directly makes sense tho

Reply:
I derived Hash impls where necessary https://github.com/gakonst/ethers-rs/pull/1964/commits/94e9415b87d21e396ca4a1d1bce38dd3519e3237
> manual hash impl for settings and including the metadatahash directly makes sense tho
Are you referring to `SettingsMetadata`?